polyfactory/value_generators/constrained_collections.py

```diff
@@ -39,6 +40,11 @@ def handle_constrained_collection(
     min_items = abs(min_items if min_items is not None else (max_items or 0))
     max_items = abs(max_items if max_items is not None else min_items + 1)
 
+    if isinstance(field_meta.annotation, EnumMeta):
```

Comment:
We should also be checking against `Literal` types as well since they can also end up in an infinite loop.